assert/assertions.go

```diff
@@ -620,6 +620,39 @@ func NotNil(t TestingT, object interface{}, msgAndArgs ...interface{}) bool {
 	return Fail(t, "Expected value not to be nil.", msgAndArgs...)
 }
 
+// NotNilThenSetNil asserts that the specified object is not zero and then sets it to zero.
+//
+//	assert.NotZeroThenSetZero(t, &x)
+func NotZeroThenSetZero[X any](t TestingT, x *X) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+
+	if ok := NotZero(t, *x); !ok {
+		return false
+	}
+
+	var zeroValue X
+	*x = zeroValue
+	return true
+}
+
+// NotNilThenSetNil asserts that the specified pointer is not nil and then sets it to nil.
+//
+//	assert.NotNilThenSetNil(t, &x)
+func NotNilThenSetNil[X any](t TestingT, x **X) bool {
```

Comment:
This func is redundant with NotZeroThenSetZero.

1. Why 2 funcs?
2. Why duplicate the implementation instead of wrapping one with the other?